### ISSUE TITLE
yield should not be a valid optional name for a generator expression

### ISSUE THREAD

**Issue by ariya** [eslint/espree#178]
Test case:

```js
espree.parse('(function *yield() {})', { ecmaFeatures: { generators: true } })
```

Actual: a syntax tree.

Expected: an error or an exception.

**Comment by nzakas** [comment]
Ah interesting, thanks!

**Comment by nzakas** [comment]
Fixed in 3.0.0.
